x/superfluid/keeper/migrate.go

```diff
@@ -10,29 +10,30 @@ import (
 	cltypes "github.com/osmosis-labs/osmosis/v15/x/concentrated-liquidity/types"
 	gammtypes "github.com/osmosis-labs/osmosis/v15/x/gamm/types"
 	lockuptypes "github.com/osmosis-labs/osmosis/v15/x/lockup/types"
+	"github.com/osmosis-labs/osmosis/v15/x/superfluid/types"
 )
 
 // RouteLockedBalancerToConcentratedMigration routes the provided lock to the proper migration function based on the lock status.
-// If the lock is superfluid delegated, it will undelegate the superfluid position and redelegate it as a concentrated liquidity position.
+// If the lock is superfluid delegated, it will instantly undelegate the superfluid position and redelegate it as a concentrated liquidity position.
 // If the lock is superfluid undelegating, it will instantly undelegate the superfluid position and redelegate it as a concentrated liquidity position, but continue to unlock where it left off.
 // If the lock is locked or unlocking but not superfluid delegated/undelegating, it will migrate the position and either start unlocking or continue unlocking where it left off.
 // Errors if the lock is not found, if the lock is not a balancer pool lock, or if the lock is not owned by the sender.
 func (k Keeper) RouteLockedBalancerToConcentratedMigration(ctx sdk.Context, sender sdk.AccAddress, lockId uint64, sharesToMigrate sdk.Coin, tokenOutMins sdk.Coins) (positionId uint64, amount0, amount1 sdk.Int, liquidity sdk.Dec, joinTime time.Time, poolIdLeaving, poolIdEntering, gammLockId, concentratedLockId uint64, err error) {
 	// Validate and retrieve pertinent data required for migration
-	poolIdLeaving, poolIdEntering, concentratedPool, preMigrationLock, remainingLockTime, synthLockBeforeMigration, isSuperfluidBonded, isSuperfluidUnbonding, err := k.prepareMigration(ctx, sender, lockId, sharesToMigrate)
+	synthLocksBeforeMigration, isSuperfluidBonded, isSuperfluidUnbonding, err := k.routeMigration(ctx, sender, lockId, sharesToMigrate)
```

Comment:
Can we instead have this return an enum instead of multiple booleans?

Reply:
High level idea:

```go
type Status int

const (
    Pending Status = iota
    Approved
    Rejected
    OnHold
)

func main() {
    var status Status = Approved

    switch status {
    case Pending:
        fmt.Println("Status: Pending")
    case Approved:
        fmt.Println("Status: Approved")
    case Rejected:
        fmt.Println("Status: Rejected")
    case OnHold:
        fmt.Println("Status: On Hold")
    }
}

```

Reply:
Good suggestion on this, was way cleaner the way you suggested. Added here https://github.com/osmosis-labs/osmosis/pull/5160/commits/9fb6fcdeeae231b72bb4a597e07a4a58f931cbba